src/rust/entrystore/src/seg/memcache.rs

```diff
@@ -31,25 +35,36 @@ impl MemcacheStorage for Seg {
             }
         }
 
+        let elapsed = now.elapsed();
+        if elapsed > Duration::from_micros(500) {
+            debug!("slow get: {:?}", elapsed);
+        }
+
         items.into_boxed_slice()
     }
 
     fn set(&mut self, entry: &MemcacheEntry) -> Result<(), MemcacheStorageError> {
+        let now = Instant::now();
         let ttl = if entry.ttl() == Some(0) {
             return Err(MemcacheStorageError::NotStored);
         } else {
             CoarseDuration::from_secs(entry.ttl().unwrap_or(0))
         };
 
-        match self.data.insert(
+        let result = match self.data.insert(
             entry.key(),
             entry.value().unwrap_or(b""),
             Some(&entry.flags().to_be_bytes()),
             ttl,
         ) {
             Ok(_) => Ok(()),
             Err(_) => Err(MemcacheStorageError::NotStored),
+        };
+        let elapsed = now.elapsed();
+        if elapsed > Duration::from_micros(500) {
```

Comment:
I would prefer if the threshold for slow reqeusts is set with other configs related to debug.